lib/engines/content_block_manager/app/models/content_block_manager/content_block/document/document_filter.rb

```diff
@@ -22,23 +46,25 @@ def is_date_present?(date_key)
       @filters[date_key].present? && @filters[date_key].all? { |_, value| value.present? }
     end
 
+    def date_from_filters(date_key)
+      filter = @filters[date_key]
+      year = filter["1i"].to_i
+      month = filter["2i"].to_i
+      day = filter["3i"].to_i
+      { year:, month:, day: }
+    end
+
     def from_date
       @from_date ||= if is_date_present?(:last_updated_from)
-                       filter = @filters[:last_updated_from]
-                       year = filter["1i"].to_i
-                       month = filter["2i"].to_i
-                       day = filter["3i"].to_i
-                       Time.zone.local(year, month, day)
+                       date = date_from_filters(:last_updated_from)
+                       Time.zone.local(date[:year], date[:month], date[:day])
                      end
     end
 
     def to_date
       @to_date ||= if is_date_present?(:last_updated_to)
-                     filter = @filters[:last_updated_to]
-                     year = filter["1i"].to_i
-                     month = filter["2i"].to_i
-                     day = filter["3i"].to_i
-                     Time.zone.local(year, month, day).end_of_day
+                     date = date_from_filters(:last_updated_to)
+                     Time.zone.local(date[:year], date[:month], date[:day]).end_of_day
```

Comment:
I added the call to `end_of_day` rather than using `Time.zone.local` with the time specifically set. Sorry, should have been a suggestion!